### DISCUSSION TITLE
Hide step in according screen breakdown

### DISCUSSION THREAD

**Discussion by vishwakarmanikhil** [gilbarbara/react-joyride#826] [Q&A] [answered]
Hey, Can we hide the particular step in mobile view? Or if I want to hide certain steps based on screen breakdown. How can I achieve that ? I am hiding certain steps in mobile I have handled that also if step taget_not_visible or target_not_found moved to next or prev based on action. but it is giving me an error on the screen.

Unhandled Rejection (TypeError): Cannot read properties of null (reading 'isConnected')

Is there any way to achieve this.

**Comment by gilbarbara** [comment, marked as answer]
hey @vishwakarmanikhil 

You can filter out the steps that aren't visible based on the screen size.

**Reply by vishwakarmanikhil** [reply]
Hi, @gilbarbara 
First of all, Thank you, You have made such a wonderful library and given your time to answer my query. Do you have any reference example for this how can I filter the steps ?

**Reply by gilbarbara** [reply]
Using [Array.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) in the `steps` (and probably `useMemo` to avoid the re-calculation for every render)

**Reply by vishwakarmanikhil** [reply]
@gilbarbara Thank you. I really appreciated it.
